### PR TITLE
Dan/2023/01/replace eth with native

### PR DESCRIPTION
**Pull request by DJViau**
Addresses https://gist.github.com/0age/349aef9a83d82b1bf2438937001adcef#replace-eth-with-native-token

### CODE REVIEW

File: contracts/interfaces/ConsiderationEventsAndErrors.sol [hunk 1]
```diff
@@ -137,15 +137,15 @@ interface ConsiderationEventsAndErrors {
     );
 
     /**
-     * @dev Revert with an error when insufficient ether is supplied as part of
-     *      msg.value when fulfilling orders.
+     * @dev Revert with an error when insufficient native token is supplied as
+     *      part of msg.value when fulfilling orders.
      */
-    error InsufficientEtherSupplied();
+    error InsufficientNativeTokenSupplied();
```
Comment (0age) [inline contracts/interfaces/ConsiderationEventsAndErrors.sol:143]: I feel like this error should be `InsufficientNativeTokensSupplied` (so plural)

Comment (DJViau) [reply contracts/interfaces/ConsiderationEventsAndErrors.sol:143]: Roger that, will update

File: reference/lib/ReferenceBasicOrderFulfiller.sol [hunk 2]
```diff
@@ -841,8 +841,8 @@ contract ReferenceBasicOrderFulfiller is ReferenceOrderValidator {
         uint256 amount,
         BasicOrderParameters calldata parameters
     ) internal {
-        // Put ether value supplied by the caller on the stack.
-        uint256 etherRemaining = msg.value;
+        // Put native token value supplied by the caller on the stack.
+        uint256 nativeTokenRemaining = msg.value;
```
Comment (0age) [inline reference/lib/ReferenceBasicOrderFulfiller.sol:845]: `nativeTokensRemaining`

File: reference/lib/ReferenceBasicOrderFulfiller.sol [hunk 3]
```diff
@@ -851,36 +851,39 @@ contract ReferenceBasicOrderFulfiller is ReferenceOrderValidator {
...
-            revert InsufficientEtherSupplied();
+        // Ensure that sufficient native token is still available.
+        if (amount > nativeTokenRemaining) {
+            revert InsufficientNativeTokenSupplied();
         }
 
-        // Transfer Ether to the offerer.
-        _transferEth(parameters.offerer, amount);
+        // Transfer native token to the offerer.
+        _transferNativeToken(parameters.offerer, amount);
```
Comment (0age) [inline reference/lib/ReferenceBasicOrderFulfiller.sol:878]: `_transferNativeTokens`

File: contracts/interfaces/ConsiderationEventsAndErrors.sol [hunk 4]
```diff
@@ -137,15 +137,15 @@ interface ConsiderationEventsAndErrors {
     );
 
     /**
-     * @dev Revert with an error when insufficient ether is supplied as part of
-     *      msg.value when fulfilling orders.
+     * @dev Revert with an error when insufficient native token is supplied as
```
Comment (0age) [inline contracts/interfaces/ConsiderationEventsAndErrors.sol:140]:
```suggestion
     * @dev Revert with an error when insufficient native tokens are supplied as
```

File: reference/lib/ReferenceBasicOrderFulfiller.sol [hunk 5]
```diff
@@ -851,36 +851,39 @@ contract ReferenceBasicOrderFulfiller is ReferenceOrderValidator {
...
             );
 
-            // Read ether amount to transfer to recipient and place on stack.
+            // Read native token amount to transfer to recipient and place on stack.
             uint256 additionalRecipientAmount = additionalRecipient.amount;
 
-            // Ensure that sufficient Ether is available.
-            if (additionalRecipientAmount > etherRemaining) {
-                revert InsufficientEtherSupplied();
+            // Ensure that sufficient native token is available.
```
Comment (0age) [inline reference/lib/ReferenceBasicOrderFulfiller.sol:857]:
```suggestion
            // Ensure that sufficient native tokens are available.
```

**Pull request merged**